src/components/EditSnack/EditSnack.js

```diff
@@ -1,26 +1,55 @@
-import React, {Component} from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import ButtonList from '../ButtonList/ButtonList';
+import ListSelection from '../listSelection/ListSelection';
 import Logo from '../Logo/Logo';
+import './EditSnack.css';
 
-class EditSnack extends Component {
-	handleClick = (id, name) => {
-		this.props.setActualItem(id);
-		this.props.history.push('/slackname');
-	};
+const importImages = require.context('./assets', true, /\.svg$/);
+const imgFilesObject = importImages.keys().reduce((images, key) => {
+	images[key] = importImages(key);
+	return images;
+}, {});
 
-	render() {
-		return (
-			<div>
-				<Logo />
-				<ButtonList
-					items={this.props.items}
-					onClick={this.handleClick}
-				/>
+const EditSnack = props => {
+	const items = props.items.map(item => {
+		const relativeImagePath = './' + item.image;
+		let actualImagePath;
+
+		if (imgFilesObject[relativeImagePath]) {
+			actualImagePath =
+				'./' + item.image + (item.image.endsWith('.svg') ? '' : '.svg');
+		} else {
+			actualImagePath = './misc-bar.svg';
+		}
+
+		const image = importImages(actualImagePath);
+		return {
+			...item,
+			image
+		};
+	});
+	return (
+		<div>
+			<Logo />
+			<div className="textinfo">
+				<nobr>
+					Sorry, I can’t recognise that snack. <br /> Please select it
+					below
+				</nobr>
 			</div>
-		);
-	}
-}
+			<ListSelection
+				items={items}
+				onClick={id => {
+					props.setActualItem(id);
```

Comment:
To be consistent with the other components it'd be good to extract this to it's own method (would need this to become a class component)